### ISSUE TITLE
[BUG] InternalExecutionError on Static data member

### ISSUE THREAD

**Issue by ChrisHobackExp** [forcedotcom/sfdx-scanner#1177]
**Describe the bug**
When running the scan `sfdx scanner:run:dfa --format=html --outfile=CodeAnalyzerDFA.html --category=Security --projectdir "./src/classes" --target "./src/classes"` we are receiving an InternalExecutionError:
`
Graph Engine identified your source and sink, but you must manually verify that you have a sanitizer in this path. Then, add an engine directive to skip the path. Next, create a Github issue for the Code Analyzer team that includes the error and stack trace. After we fix this issue, check the Code Analyzer release notes for more info. Error and stacktrace: UnexpectedException: FieldDeclarationStatements{properties={FirstChild=false, BeginLine=14, DefiningType_CaseSafe=mystringhelper, LastChild=true, DefiningType=MyStringHelper, EndLine=14, childIdx=3, BeginColumn=12}}: com.salesforce.graph.symbols.PathScopeVisitor.visit(PathScopeVisitor.java:1142);com.salesforce.graph.symbols.DefaultSymbolProviderVertexVisitor.visit(DefaultSymbolProviderVertexVisitor.java:409);com.salesforce.graph.vertex.FieldDeclarationStatementsVertex.visit(FieldDeclarationStatementsVertex.java:25);com.salesforce.graph.ops.expander.ApexPathExpander.shouldVisitChildren(ApexPathExpander.java:573);com.salesforce.graph.ops.expander.ApexPathExpander.lambda$visit$1(ApexPathExpander.java:519);java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
`
on the following class:
```
global virtual class MyStringHelper
{
    global static final String NEW_LINE;

    static {

    }

    global static Boolean isEmpty(String val) {
        return (val == null || val.trim().length() == 0);
    }

    static Map<String, String> prefixedNameCache = new Map<String, String>();
}
```

While this is not the explicit class definition we use in our product, I have slimmed down and stripped out most of the definition that still gives the same error.

_Please note that we are waiting for this to be resolved before we can submit our app for security review._

**To Reproduce**
Run the above command on the supplied class to get this error. Please note that I have actually seen that at times it would report back with no issues. In such a case please run again.

**Expected behavior**
It seems like there is no FLS issue here. I would expect this file to be clean and no violations reported.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Scanner Version 3.13.0
 - sfdx 7.209.6
 - cci 3.79.0

**Additional context**
The provided class is obviously a mock of the original class. If the original class would be more helpful, this can be emailed or provided via another delivery method. We receive this InternalExecutionError across all classes calling this class. Additionally, the ignore directives do not work for this type of Error.

"Workaround": None

"Urgency": Preventing security review

**Comment by jfeingold35** [comment]
@ChrisHobackExp , thanks for bringing this to our attention. We'll log it as a bug and start investigating. In the meantime, I was able to resolve the error by moving the definition of `prefixedNameCache` to line 4, above the static block.
Please verify that this works for you. If so, we can update the name of the Issue accordingly.

**Comment by ChrisHobackExp** [comment]
@jfeingold35 thanks for getting back to me so fast!

Unfortunately, this does not work in our Ap's version of the classp. While the error mentioned in the issue goes away, a much more obfuscated error appears:

```
Graph Engine identified your source and sink, but you must manually verify that you have a sanitizer in this path. Then, add an engine directive to skip the path. Next, create a Github issue for the Code Analyzer team that includes the error and stack trace. After we fix this issue, check the Code Analyzer release notes for more info. Error and stacktrace: StackOverflowError: null: java.base/java.util.Stack.peek(Stack.java:103);com.salesforce.graph.symbols.PathScopeVisitor.getApexValue(PathScopeVisitor.java:325);com.salesforce.graph.symbols.PathScopeVisitor.getApexValue(PathScopeVisitor.java:340);com.salesforce.graph.symbols.PathScopeVisitor.getApexValue(PathScopeVisitor.java:340);com.salesforce.graph.symbols.PathScopeVisitor.getApexValue(PathScopeVisitor.java:340);com.salesforce.graph.symbols.PathScopeVisitor.getValue(PathScopeVisitor.java:209)
```

I have a smaller version of our class that still causes this newer issue, however, the class still contains some business logic and I would prefer to deliver this through email (I don't think I can remove anything more from the class). Feel free to email me at Chris.Hoback@exponentpartners.com and I can reply with the class.

It is possible that this new error is not directly related to the same line in question. I have removed the data member declaration and the violation still shows up. But then again, maybe there is a similar underlying issue causing both of these violations.

Thanks!

**Comment by jfeingold35** [comment]
@ChrisHobackExp , the StackOverflow error has been logged as its own issue, see #1178.
We'll keep you posted and let you know once we have fixes or workarounds.
We appreciate your patience.

**Comment by ChrisHobackExp** [comment]
hey @jfeingold35, Just looking for an update or any new information on this issue. We will be going through security review soon-ish. I assume the security review team is aware of this issue and won't ding us for it, but also wanted to know if there was any additional guidance you would recommend regarding this error or handling of it.

**Comment by jfeingold35** [comment]
Unfortunately, no new information at this time. Thank you for your patience. We'll do our best to get you some information in a timely manner.

**Comment by stephen-carter-at-sf** [comment]
Marking this as a duplicate of https://github.com/forcedotcom/sfdx-scanner/issues/1497
